code/client/munkilib/FileRepo.py

```diff
@@ -0,0 +1,280 @@
+    NETFSMOUNTURLSYNC_AVAILABLE = True
+except (ImportError, KeyError):
+    NETFSMOUNTURLSYNC_AVAILABLE = False
+
+if NETFSMOUNTURLSYNC_AVAILABLE:
+    class ShareMountException(Exception):
+        '''An exception raised if share mounting failed'''
+        pass
+
+
+    class ShareAuthenticationNeededException(ShareMountException):
+        '''An exception raised if authentication is needed'''
+        pass
+
+
+    def mount_share(share_url):
+        '''Mounts a share at /Volumes, returns the mount point or raises an
+        error'''
+        sh_url = CFURLCreateWithString(None, share_url, None)
+        # Set UI to reduced interaction
+        open_options = {NetFS.kNAUIOptionKey: NetFS.kNAUIOptionNoUI}
+        # Allow mounting sub-directories of root shares
+        mount_options = {NetFS.kNetFSAllowSubMountsKey: True}
+        # Mount!
+        result, output = NetFS.NetFSMountURLSync(
+            sh_url, None, None, None, open_options, mount_options, None)
+        # Check if it worked
+        if result != 0:
+            if result in (errno.ENOTSUP, errno.EAUTH):
+                # errno.ENOTSUP is returned if an afp share needs a login
+                # errno.EAUTH is returned if authentication fails (SMB for sure)
+                raise ShareAuthenticationNeededException()
+            raise ShareMountException(
+                'Error mounting url "%s": %s, error %s'
+                % (share_url, os.strerror(result), result))
+        # Return the mountpath
+        return str(output[0])
+
+
+    def mount_share_with_credentials(share_url, username, password):
+        '''Mounts a share at /Volumes, returns the mount point or raises an
+        error. Include username and password as parameters, not in the
+        share_path URL'''
+        sh_url = CFURLCreateWithString(None, share_url, None)
+        # Set UI to reduced interaction
+        open_options = {NetFS.kNAUIOptionKey: NetFS.kNAUIOptionNoUI}
+        # Allow mounting sub-directories of root shares
+        mount_options = {NetFS.kNetFSAllowSubMountsKey: True}
+        # Mount!
+        result, output = NetFS.NetFSMountURLSync(
+            sh_url, None, username, password, open_options, mount_options, None)
+        # Check if it worked
+        if result != 0:
+            raise ShareMountException(
+                'Error mounting url "%s": %s, error %s'
+                % (share_url, os.strerror(result), result))
+        # Return the mountpath
+        return str(output[0])
+
+
+    def mount_share_url(share_url):
+        '''Mount a share url under /Volumes, prompting for password if needed
+        Raises ShareMountException if there's an error'''
+        try:
+            mount_share(share_url)
+        except ShareAuthenticationNeededException:
+            username = raw_input('Username: ')
+            password = getpass.getpass()
+            mount_share_with_credentials(share_url, username, password)
+
+class FileRepo:
```

Comment:
Any reason this isn't `class FileRepo(object):` -- as is this defines an "Old-style class". Is that what you intended? If so, why?